File: kraken/src/main/java/glitch/kraken/object/json/Channel.java
package glitch.kraken.object.json;

import com.google.gson.annotations.SerializedName;
import glitch.api.objects.enums.BroadcasterType;
import glitch.kraken.object.json.interfaces.IUser;
import lombok.Data;

import java.awt.*;
import java.time.Instant;
import java.util.Locale;

@Data
public class Channel implements IUser {
    @SerializedName(value = "id", alternate = "_id")
    private final Long id;
    @SerializedName(value = "username", alternate = {"login", "name"})
    private final String username;
    private final String displayName;

    private final Instant createdAt;
    private final Instant updatedAt;

    private final String logo;

    @SerializedName(value = "bio", alternate = {"description"})
    private final String userBio;
    private final boolean mature;
    @SerializedName("status")
    private final String title;
    private final Locale broadcasterLanguage;
    private final String broadcastSoftware;
    private final String game;
    private final Locale language;
    private final boolean partner;
    private final String videoBanner;
    private final String profileBanner;
    private final Color profileBannerBackgroundColor;
    private final String url;
    private final long views;
    private final long followers;
    private final BroadcasterType broadcasterType;
    private final boolean privateVideo;
    private final boolean privacyOptionsEnabled;
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/channel.kt
package glitch.kraken.`object`.json

import com.google.gson.annotations.SerializedName
import glitch.api.objects.adapters.SerializeTo
import glitch.api.objects.enums.BroadcasterType
import glitch.api.objects.json.interfaces.Creation
import glitch.api.objects.json.interfaces.IDObject
import glitch.api.objects.json.interfaces.Updated
import glitch.kraken.`object`.json.impl.ChannelImpl
import sun.net.www.protocol.http.AuthCacheImpl
import java.awt.Color
import java.time.Instant
import java.util.*

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
@SerializeTo(ChannelImpl::class)
interface Channel : IDObject<Long>, Creation, Updated {
    @get:SerializedName(value = "username", alternate = ["login", "name"])
    val username: String
    @get:SerializedName("status")
    val title: String
    val broadcasterLanguage: Locale
    val displayName: String
    val followers: Long
    val game: String
    val language: Locale
    val logo: String
    val mature: Boolean
    val partner: Boolean
    val profileBanner: String?
    val profileBannerBackgroundColor: Color?
    val url: String
    val videoBanner: String?
    val views: Long
}

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
@SerializeTo(AuthCacheImpl::class)
interface AuthChannel : Channel {
    val broadcasterType: BroadcasterType
    val email: String
    val streamKey: String
}

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class ChannelUserFollow(
        @SerializedName("user")
        override val data: User,
        @SerializedName("notifications")
        override val hasNotifications: Boolean,
        override val createdAt: Instant
) : Follow<User>

data class CommercialData(
        @SerializedName("Length")
        val length: Int,
        @SerializedName("Message")
        val message: String,
        @SerializedName("RetryAfter")
        val retryAfter: Int
)